StaticAnalyzer/views/android/static_analyzer.py

```diff
@@ -508,3 +524,27 @@ def gen_downloads(app_dir, md5, icon_path=''):
 
     except:
         PrintException("Generating Downloads")
+
+
+def get_app_name(app_path, app_dir, tools_dir, is_apk):
+    """ get app name """
+    
+    data = ""
+    if is_apk:
+        output_dir = os.path.join(app_dir, "apktool_out")
+        strings_file = os.path.join(output_dir, "res/values/strings.xml")
+    else:
+        if os.path.exists('app/src/main/res/values/strings.xml'):
+            strings_file = os.path.join(app_dir, 'app/src/main/res/values/strings.xml')
+        else:
+            strings_file = os.path.join(app_dir, 'res/values/strings.xml')
```

Comment:
What if the path doesn't exist? `else` may not be a good idea.

Reply:
The pseudo logic may be like
if apk and the strings.xml exists 
 -> set path
else if studio projects strings.xml path exist
  -> set path
else if eclipse project strings.xml path exist
  -> set path
else 
   return an empty string # couldn't find the real app name, also doesn't break the further processing.